Intermittent ECONNRESET errors

### Environment

Node version: 11.12
Npm version: 6.2
OS and version: Windows 2018 server (Azure DevOps release pipelines)
azure-devops-node-api version: 7.0.0 (also 6.x)

### Issue Description

I'm experiencing intermittent [ECONNRESET](https://stackoverflow.com/questions/17245881/node-js-econnreset) errors [running](https://github.com/dmitryserbin/azdev-release-orchestrator) various Build/Core/Release API methods against Azure DevOps projects or pipelines. Failure rate is about 1-2 per 100 calls.

The issue affects various APIs, for example:

- IReleaseApi > getRelease, getReleases, getReleaseDefinition, etc.
- IBuildApi > getBuild, getBuilds, getBuildTags, etc.
- ICoreApi > getProject

Basically, I'm forced to use a retry mechanism on top of the methods as a workaround:

```typescript
        let result: Build | undefined;
        let retryAttempt: number = 0;

        while (retryAttempt < retry) {
            try {
                retryAttempt++;
                // Get build
                result = await this.buildApi.getBuild(buildId, projectName);
                // Stop retrying on success
                retryAttempt = retry;
            } catch {
                console.log(`Retry retrieving build..`);
                // Delay before next retry
                await this.delay(timeout);
            }
        }
```

I do not believe there's anything wrong with your code, but please advise if it's possible to address these type of issues on the library level? For example, by introducing a native retry mechanism (or anything else you could recommend).

### Expected behaviour

API should return response.

### Actual behaviour

ECONNRESET thrown.

### Steps to reproduce

It is challenging to reproduce as you need need to be very lucky. The task I'm running in a pipeline is making >100 call per ~5 min. I have never experienced this issue locally - it only happens in Azure DevOps pipelines.

### Logs

Not much logs at this point, but I can get some, if needed.

```
2019-03-16T16:06:07.4473715Z ##[error]read ECONNRESET
```

--- a/api/WebApi.ts
+++ b/api/WebApi.ts
@@ -233,9 +233,12 @@ export class WebApi {
 
     // TODO: Don't call resource area here? Will cause infinite loop?
     public async getLocationsApi(serverUrl?: string, handlers?: VsoBaseInterfaces.IRequestHandler[]): Promise<locationsm.ILocationsApi> {
+        let optionsClone: VsoBaseInterfaces.IRequestOptions = Object.assign({}, this.options);
+        optionsClone.allowRetries = true;
+        optionsClone.maxRetries = 5;
         serverUrl = await serverUrl || this.serverUrl;
         handlers = handlers || [this.authHandler];
-        return new locationsm.LocationsApi(serverUrl, handlers, this.options);
+        return new locationsm.LocationsApi(serverUrl, handlers, optionsClone);
     }
 
     public async getNotificationApi(serverUrl?: string, handlers?: VsoBaseInterfaces.IRequestHandler[]): Promise<notificationm.INotificationApi> {

--- a/api/interfaces/common/VsoBaseInterfaces.ts
+++ b/api/interfaces/common/VsoBaseInterfaces.ts
@@ -85,10 +85,12 @@ export interface IHttpClientResponse {
 }
 
 export interface IRequestOptions {
-    socketTimeout?: number,
-    ignoreSslError?: boolean,
-    proxy?: IProxyConfiguration
-    cert?: ICertConfiguration
+    socketTimeout?: number;
+    ignoreSslError?: boolean;
+    proxy?: IProxyConfiguration;
+    cert?: ICertConfiguration;
+    allowRetries?: boolean;
+    maxRetries?: number;
 }
 
 export interface IProxyConfiguration {

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "azure-devops-node-api",
     "description": "Node client for Azure DevOps and TFS REST APIs",
-    "version": "7.1.0",
+    "version": "7.2.0",
     "main": "./WebApi.js",
     "types": "./WebApi.d.ts",
     "scripts": {
@@ -26,7 +26,7 @@
     "dependencies": {
         "os": "0.1.1",
         "tunnel": "0.0.4",
-        "typed-rest-client": "1.0.9",
+        "typed-rest-client": "1.2.0",
         "underscore": "1.8.3"
     },
     "devDependencies": {

